src/tools/gdb/plan_dump.py

```diff
@@ -0,0 +1,605 @@
+				motion_recv = 1
+			else:
+				sname = "Gather Motion"
+				motion_recv = 1
+		elif typ == Motion.MOTIONTYPE_EXPLICIT:
+			sname = "Explicit Redistribute Motion"
+			# motion_recv = getgpsegmentCount() it's not easy to implement analog for core dump
+
+		if self.__pstmt["planGen"] == PLANGEN_PLANNER:
+			slice = self.__currentSlice
+			if bool(slice["directDispatch"]["isDirectDispatch"]) == True:
+				# Special handling on direct dispatch
+				motion_snd = List.list_length(slice["directDispatch"]["contentIds"])
+			elif self.__plan["lefttree"]["flow"]["flotype"] == Flow.FLOW_SINGLETON:
+				# For SINGLETON we always display sender size as 1
+				motion_snd = 1
+			else:
+				# Otherwise find out sender size from outer plan
+				motion_snd = int(self.__plan["lefttree"]["flow"]["numsegments"])
+
+			if self.__mot["motionType"] == Motion.MOTIONTYPE_FIXED and bool(self.__mot["isBroadcast"]) == False:
+				# In Gather Motion always display receiver size as 1
+				motion_recv = 1
+			else:
+				# Otherwise find out receiver size from plan
+				motion_recv = int(self.__plan["flow"]["numsegments"])
+
+		return '%s %d:%d %s' % (sname, motion_snd, motion_recv, show_dispatch_info(self.__currentSlice, self.__plan, self.__pstmt))
+
+class ShareInputScan(object):
+	gdb_type = gdb.lookup_type("ShareInputScan").pointer()
+
+	def __init__(self, node, currentSlice):
+		self.__sics = node.cast(ShareInputScan.gdb_type)
+		self.__currentSlice = currentSlice
+
+	def print_node(self):
+		slice_id = -1
+		if self.__currentSlice != 0:
+			slice_id = int(self.__currentSlice["sliceIndex"])
+
+		return "ShareInputScan (share slice:id %d:%d)" % (slice_id, int(self.__sics["share_id"]))
+
+class PartitionSelector(object):
+	gdb_type = gdb.lookup_type("PartitionSelector").pointer()
+
+	def __init__(self, node, reloidMap):
+		self._ps = node.cast(PartitionSelector.gdb_type)
+		self.__reloidMap = reloidMap
+
+	def print_node(self):
+		relname = self.__reloidMap[int(self._ps["relid"])]
+		return "PartitionSelector for %s (dynamic scan id: %d)" % (relname, int(self._ps["scanId"]))
+
+class Scan(object):
+	gdb_type = gdb.lookup_type("Scan").pointer()
+	__typ = ""
+
+	def __init__(self, node, rtableMap, typ):
+		self.scan = node.cast(Scan.gdb_type)
+		self.__rtableMap = rtableMap
+		self.__typ = typ
+
+	def print_node(self, indexid):
+		indexInfo = ""
+		if indexid is not None:
+			indexInfo = " (used index, indexoid %s)" % str(indexid)
+		id = int(self.scan["scanrelid"])
+		relanme = self.__rtableMap[id]
+		return "%s on %s%s" % (self.__typ, relanme, indexInfo)
+
+class PlanDumperCmd(gdb.Command):
+	"""Print the plan nodes like pg explain"""
+
+	def __cleanup__(self):
+		self.__result = ""
+		self.__tabCnt = 0
+		self.__curPlanName = None
+		self.__currentSlice = 0
+		self.__pstmt = None
+		self.__state = None
+		self.__rtableMap = {}
+		self.__reloidMap = {}
+
+	def __init__(self):
+		super(PlanDumperCmd, self).__init__("plan_dump_cmd", gdb.COMMAND_USER, gdb.COMPLETE_SYMBOL)
```

Comment:
The script name is plan_dump.py, so what about renaming the command to "plan_dump" too?